.github/workflows/push.yml

```diff
@@ -0,0 +1,57 @@
+name: Single Build & Test (Push)
+
+# workflow is for a branch push only and ignores master.
+# push to master (which is also pull requet merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+  workflow_dispatch:
+    inputs: 
+      node:
+        required: false
+        description: 'A node version (e.g 10.16.0)'
+        default: '14' # TODO: update to 16 when support is added
+
+jobs:
+  build-single-test:
+    runs-on: ubuntu-latest # environment job will run in
+
+    env:
+      # TODO: move to staging when AO-20147
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }} 
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Setup Node ${{ github.event.inputs.node || github.event.inputs.node.default }}
```

Comment:
This will not work unless your workflow is triggered manually. In case of a push trigger, both of these parameters will be empty.

Reply:
See blow.

Reply:
Hmm, the naming does not work properly on the run you pointed out below:
![Screen Shot 2021-07-09 at 12 57 20 PM](https://user-images.githubusercontent.com/60111015/125130150-5c77f580-e0b5-11eb-93d1-2f584352285b.png)

Reply:
For your reference, I tried this out in a test repo, and the same behaviour there, you will see what version of Node was used when run manually, but no version will be displayed when run through the push trigger.
![Screen Shot 2021-07-09 at 12 57 20 PM](https://user-images.githubusercontent.com/60111015/125130308-a06afa80-e0b5-11eb-8cdd-fb10630ae1a1.png)

Reply:
Yah. it (was) `false`.